### DISCUSSION TITLE
Camera - get actual (permanent) path of picture from gallery, not cached one.

### DISCUSSION THREAD

**Discussion by T0BiD** [ionic-team/capacitor#4306] [Q&A] [answered]
Hi,

I want people to see certain (selected) images from their gallery/photos in my App.

I'm using the Capacitor Camera plugin to let them select the image:

```
const capturedPhoto = await Camera.getPhoto({
        resultType: CameraResultType.Uri,
        source: CameraSource.Photos,
        quality: 100,
      });
```

When I get the path from the capturedPhoto it's  a path to the cached image, not the 'real' one.
`capturedPhoto.path: file:///data/user/0/<appname>/cache/307474195.1615066695588.jpeg`

This path is not permanent. I could move the image to another location (data) for permanent storage, but do I really want to store all the pictures they have already saved again? That seems like a massive waste of storage. Especially when we're talking tens to hundreds of images.

All the tutorials on using Capacitor Camera are actually doing this, but they're usually taking a new picture where this kinda makes sense. But in my case it's not a smart solution.

Is there any way to actually get the real path of the image?

**Comment by jcesarmobile** [comment, marked as answer]
Images that don’t belong to the app can only be accessed through content urls, the app needs permission to access those urls and the permission it’s only temporary. In some android versions you can still convert the content url to a file path and access it freely, but google makes it harder on every new android OS release.

**Reply by T0BiD** [reply]
So what you're saying is that it's basically not possible on most android versions (or feasible in the long run) and I should store the pictures a second time?  
Just a simple yes will suffice :D

**Reply by jcesarmobile** [reply]
It’s not possible unless you want to work with the content urls directly

**Comment by dotNetkow** [comment]
Maybe the Capacitor community [media plugin](https://github.com/capacitor-community/media) would help?
